### PR TITLE
[SLS-2545] Internal filtering of DNS, TCP traces

### PR DESCRIPTION
**Pull request by SiddhithaBhoopathy**
### What does this PR do?

Filters out TCP traces from local host address 127.0.0.1 and DNS traces from non-routable address 0.0.0.0 and local host address 127.0.0.1.
Example traces:
APM Traces Datadog 2023-01-31 at 11 28 15 a m
APM Traces Datadog 2023-01-31 at 11 28 58 a m

### Motivation

https://github.com/DataDog/datadog-lambda-extension/issues/78

### CODE REVIEW

File: pkg/serverless/trace/trace.go [hunk 1]
```diff
@@ -46,6 +53,12 @@ const lambdaExtensionURLPrefix = "http://127.0.0.1:8124"
 // lambdaStatsDURLPrefix is the first part of a URL for a call from Statsd
 const lambdaStatsDURLPrefix = "http://127.0.0.1:8125"
 
+// LocalHostNametPrefix is the first part of a URL from the localhost address for both DNS and TCP traces
+const LocalHostNamePrefix = "127.0.0.1"
```
Comment (DylanLovesCoffee) [inline pkg/serverless/trace/trace.go:57]: nit: `LocalHostNamePrefix` -> `localhostNamePrefix`

File: pkg/serverless/trace/trace.go [hunk 2]
```diff
@@ -87,6 +101,7 @@ func (s *ServerlessTraceAgent) Flush() {
 	if s.Get() != nil {
 		s.ta.FlushSync()
 	}
+	fmt.Println("Hello")
```
Comment (DylanLovesCoffee) [inline pkg/serverless/trace/trace.go:104]: ✏️

File: pkg/serverless/trace/trace.go [hunk 3]
```diff
@@ -130,9 +148,35 @@ func filterSpanFromLambdaLibraryOrRuntime(span *pb.Span) bool {
...
+	}
+
+	// Filter out TCP traces
+
+	if val, ok := span.Meta[tcpAddressMetaKey]; ok {
+		if strings.HasPrefix(val, LocalHosNamePrefix) {
+			log.Debugf("Detected TCP span with tcp adress %s, removing it", val)
+			return true
+		}
+	}
```
Comment (DylanLovesCoffee) [inline pkg/serverless/trace/trace.go:173]: At the moment, these conditions will filter out _all_ TCP + DNS spans with `127.0.0.1` or `0.0.0.0`. Let's narrow this logic down to include ports 8124 (DD extension), 8125 (dogstatsd) and 9001 (Lambda runtime).

File: pkg/serverless/trace/trace.go [hunk 4]
```diff
@@ -130,9 +148,35 @@ func filterSpanFromLambdaLibraryOrRuntime(span *pb.Span) bool {
...
+		}
+
+		if strings.HasPrefix(val, dnsHostNamePrefix) {
+			log.Debugf("Detected DNS span with non-routable address %s, removing it", val)
+			return true
+		}
+	}
+
+	// Filter out TCP traces
+
```
Comment (DylanLovesCoffee) [inline pkg/serverless/trace/trace.go:167]: ✏️

**Comment by SiddhithaBhoopathy** [comment]
Created another one - https://github.com/DataDog/datadog-agent/pull/13236

**Pull request closed without merging**